File: 0303/303.go
func rangeSumQueryImmutable(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := rangeSumQueryImmutable(root.Left)
	r := rangeSumQueryImmutable(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 0303/303.java

class NumArray {
    int[] prefix;
    public NumArray(int[] nums) {
        prefix = new int[nums.length + 1];
        for (int i = 0; i < nums.length; i++) prefix[i+1] = prefix[i] + nums[i];
    }
    public int sumRange(int left, int right) {
        return prefix[right+1] - prefix[left];
    }
}
